src/prng.c

```diff
@@ -0,0 +1,93 @@
+/******************************************************************************
+ *
+ * Copyright 2017 Xaptum, Inc.
+ * 
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ * 
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License
+ *
+ *****************************************************************************/
+
+#include <ecdaa/prng.h>
+
+#ifdef USE_LIBSODIUM_RNG_SEED_FUNCTION
+
+#include <sodium.h>
+#include <stdio.h>
+#include <stdlib.h>
+#if defined(__linux__)
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/ioctl.h>
+#include <linux/random.h>
+#include <stdio.h>
+#include <stdlib.h>
+#endif  // defined(__linux__)
+
+static ecdaa_prng_seed_function_type seed_func = &randombytes_buf;
+static int sodium_initialized = 0;
+
+static void sodium_init_if_needed(void);
+
+#else   // USE_LIBSODIUM_RNG_SEED_FUNCTION
+
+static ecdaa_prng_seed_function_type seed_func;
+
+#endif   // USE_LIBSODIUM_RNG_SEED_FUNCTION
+
+void ecdaa_prng_init(struct ecdaa_prng *prng_in)
+{
+#ifdef USE_LIBSODIUM_RNG_SEED_FUNCTION
+    sodium_init_if_needed();
```

Comment:
Can this be put in `ecdaa_init()` instead of here?  Then we don't need to track the libsodium init state.  If the user isn't using libsodium, `ecdaa_init()` will just be a no-op for now.